src/NuGet.Core/NuGet.Packaging/Signing/Archive/EndOfCentralDirectoryRecord.cs

```diff
@@ -47,5 +46,65 @@ internal static EndOfCentralDirectoryRecord Read(BinaryReader reader)
 
             return header;
         }
+
+        private static void SeekToEndOfCentralDirectoryRecord(BinaryReader reader)
+        {
+            var byteSignature = BitConverter.GetBytes(Signature);
+            var length = reader.BaseStream.Length;
+
+            if (length < byteSignature.Length)
+            {
+                ThrowByteSignatureNotFoundException(byteSignature);
+            }
+
+            const int DefaultBufferSize = 4096;
+
+            var bufferSize = (int)Math.Min(length, DefaultBufferSize);
```

Comment:
why is the int cast needed? Math.Min has an overload for int which returns an int.

Reply:
Because `length`, which is the value of [`Stream.Length`](https://msdn.microsoft.com/en-us/library/system.io.stream.length(v=vs.110).aspx), is `long`.  The C# compiler performs an implicit conversion of `DefaultBufferSize` to `long`.

This type cast is necessary because [`Stream.ReadBytes(...)`](https://github.com/NuGet/NuGet.Client/pull/2087/files/387d7480ddf9bd81f84033ac384b276cfad38f72#diff-5681f97514e9f0030f2753902a389ba1R70) requires an `int`.  Without it, it's a compile error.